src/multiplex/box.py

```diff
@@ -176,6 +177,8 @@ def activate_input_mode(self):
         return True
 
     def exit_input_mode(self):
+        if self.state.auto_collapse and self.holder.iterator.metadata["exit_code"] == 0:
```

Comment:
https://github.com/dankilman/multiplex/blob/master/src/multiplex/viewer.py#L365
would be a more appropriate place to put this, as `exit_input_mode` may be called at any point even when a process is still running.
Which brings me to the second point, `exit_code` might not be available if the iterator is not a process (for example using the `Controller` API) so you may want to use `get`